packages/nimble-components/src/table/specs/table-keyboard-navigation-hld.md

```diff
@@ -0,0 +1,180 @@
+    -   Pressing `F2` will focus the 1st/ primary interactive element, but not trigger its functionality
+    -   At this point, the table is no longer in 'navigation mode' (will not handle arrow key presses). `Tab`/`Shift-Tab` will also be handled by the active element by default, but may bubble up and be handled by the table (resulting in a focus change) if the content does not process `Tab`/`Shift-Tab` presses. To return to 'navigation mode', and focus the cell again:
+        -   `Esc` is pressed: If the user had edited content (e.g. textbox), it may be reverted.
+        -   `Enter` is pressed: If the user was editing content or interacting with a control, those changes will be committed.
+-   For all cells with action menu buttons:
+    -   `Ctrl-Enter` will trigger the action menu button and open the action menu. Note: If the action menu button is the only interactive element in the cell, `Enter` will do the same thing.
+-   Pressing `Tab` will focus the 1st/ primary interactive element in the cell. Subsequent `Tab` presses will focus subsequent interactive elements in the cell, including the action menu button. `Shift-Tab` will focus the previous interactive element. Note: `Tab` will not cycle within the cell (it will go to the next focusable elements in the row, or focus will leave the table, as outlined in the "`Tab`/`Shift-Tab` behavior" section above).
+
+If a cell is focused and `ArrowDown` or `ArrowUp` is pressed, when there is an available row in the navigation direction, focus will move to the cell in the current focused column of the new row.
+
+`Ctrl-Alt modifiers`: It appears to be a common convention for screen readers to leverage `Ctrl-Alt Arrows` to move the screen reader focus to each of the cells in a row. As such, it may be desirable for our implementation to explicitly not handle keyboard interactions that involve both modifier keys, and instead respond to any resulting focus change, and update any internal state as needed based on the newly focused element. This is what the ARIA `treegrid` example does. It's not clear whether this behavior will require special handling by us or not, but we should test this scenario.
+
+#### Row Selection
+
+If the table has `selection-mode` of `single` or `multiple`:
+
+-   If a row is focused, pressing `Space` will select/unselect that row
+-   If a row's selection checkbox is focused, pressing `Space` will toggle the checkbox, selecting/unselecting that row
+-   **(Open question)** If a single cell is focused, should pressing `Space` select/unselect that row?
+
+Since a row's selection checkbox is currently `role=gridcell`, it will be able to be navigated to with `LeftArrow`/`RightArrow`.
+
+#### Grouping
+
+Since the entire group header row is a button, the explicit expand/collapse button in that row is not keyboard focusable.
+
+If a group header row is focused, pressing `Space` or `Enter` will expand or collapse that group.
```

Comment:
What do you think about also allowing `ArrowLeft` to collapse and `ArrowRight` to expand group rows? That would feel more in line with the proposal for hierarchical parent rows. I can imagine a use case where a user wants to collapse several nearby rows with a series of left+downs in quick sequence (in fact I've done this in other apps) but that would be interrupted if some of the rows are groups.

I can't think of any conflicts that this would introduce with other gestures you describe in this document, but obviously if it does then we should drop this idea.

Reply:
Seems reasonable. Added.